### PR TITLE
Add ASCII fast path for ILIKE scalar (90% faster)

### LINKED ISSUE #3311: ILIKE Kernels Inconsistent Case Folding
**Describe the bug**

The regex crate only implements the [simple loose matching](https://docs.rs/regex/latest/regex/#unicode) as defined by the [unicode standard](https://unicode.org/reports/tr18/#Simple_Loose_Matches).

In particular the `ﬀ` ligature will not be considered to be case insensitive equal to `ff`, `Ff`, etc... `ß` will not be considered as a lower-case of `SS`, etc...

However, the Rust standard library function `to_uppercase()` implements the full unicode conversion specification. Consequently there is a mismatch in the semantics between the ilike kernels with short-circuit implementations (i.e. starts_with) and the regex versions.

**To Reproduce**

```
test_utf8_scalar!(
    test_utf8_array_ilike_unicode_ends,
    test_utf8_array_ilike_unicode_ends_dyn,
    vec![
        "sdlkdfFFooß",
        "sdlkdfFFooSS",
        "sdlkdfFFooss",
        "FFooS",
        "FFoos",
        "ﬀooSS",
        "ﬀooß"
    ],
    "%FFooSS",
    ilike_utf8_scalar,
    ilike_utf8_scalar_dyn,
    vec![true, true, true, false, false, true, true]
);

test_utf8_scalar!(
    test_utf8_array_ilike_unicode_contains,
    test_utf8_array_ilike_unicode_contains_dyn,
    vec![
        "sdlkdfFooßsdfs",
        "sdlkdfFooSSdggs",
        "sdlkdfFoosssdsd",
        "FooS",
        "Foos",
        "ﬀooSS",
        "ﬀooß"
    ],
    "%FFooSS%",
    ilike_utf8_scalar,
    ilike_utf8_scalar_dyn,
    vec![true, true, true, false, false, true, true]
);

test_utf8_scalar!(
    test_utf8_array_ilike_unicode_complex,
    test_utf8_array_ilike_unicode_complex_dyn,
    vec![
        "sdlkdfFooßsdfs",
        "sdlkdfFooSSdggs",
        "sdlkdfFoosssdsd",
        "FooS",
        "Foos",
        "ﬀooSS",
        "ﬀooß"
    ],
    "%FF__SS%",
    ilike_utf8_scalar,
    ilike_utf8_scalar_dyn,
    vec![true, true, true, false, false, true, true]
);
```

**Expected behavior**

DuckDB at least appears to only handle the simple conversion as well

```
>>> duckdb.query("Select 'ﬀooß' ILIKE 'ffooss'").fetchall()
[(False,)]
>>> duckdb.query("Select 'FFooß' ILIKE 'ffooss'").fetchall()
[(False,)]
>>> duckdb.query("Select 'FFooSS' ILIKE 'ffooss'").fetchall()
[(True,)]
```

We should handle this consistently

**Additional context**

### PR DESCRIPTION
**Pull request by tustvold**
# Which issue does this PR close?

Closes #3311 

# Rationale for this change
 

```
ilike_utf8 scalar equals
                        time:   [218.99 µs 219.05 µs 219.10 µs]
                        change: [-90.262% -90.257% -90.253%] (p = 0.00 < 0.05)
                        Performance has improved.
Found 9 outliers among 100 measurements (9.00%)
  5 (5.00%) high mild
  4 (4.00%) high severe

Benchmarking ilike_utf8 scalar contains: Warming up for 3.0000 s
Warning: Unable to complete 100 samples in 5.0s. You may wish to increase target time to 9.1s, enable flat sampling, or reduce sample count to 50.
ilike_utf8 scalar contains
                        time:   [1.8097 ms 1.8104 ms 1.8111 ms]
                        change: [-56.369% -56.326% -56.264%] (p = 0.00 < 0.05)
                        Performance has improved.
Found 9 outliers among 100 measurements (9.00%)
  3 (3.00%) low mild
  4 (4.00%) high mild
  2 (2.00%) high severe

ilike_utf8 scalar ends with
                        time:   [261.52 µs 261.58 µs 261.65 µs]
                        change: [-88.863% -88.850% -88.831%] (p = 0.00 < 0.05)
                        Performance has improved.
Found 4 outliers among 100 measurements (4.00%)
  1 (1.00%) high mild
  3 (3.00%) high severe

ilike_utf8 scalar starts with
                        time:   [265.40 µs 265.45 µs 265.50 µs]
                        change: [-88.692% -88.679% -88.656%] (p = 0.00 < 0.05)
                        Performance has improved.
Found 11 outliers among 100 measurements (11.00%)
  2 (2.00%) low mild
  6 (6.00%) high mild
  3 (3.00%) high severe

Benchmarking ilike_utf8 scalar complex: Warming up for 3.0000 s
Warning: Unable to complete 100 samples in 5.0s. You may wish to increase target time to 9.2s, enable flat sampling, or reduce sample count to 50.
ilike_utf8 scalar complex
                        time:   [1.8319 ms 1.8327 ms 1.8334 ms]
                        change: [-4.2733% -4.1030% -3.9344%] (p = 0.00 < 0.05)
                        Performance has improved.
Found 4 outliers among 100 measurements (4.00%)
  3 (3.00%) high mild
  1 (1.00%) high severe

nilike_utf8 scalar equals
                        time:   [232.53 µs 232.59 µs 232.65 µs]
                        change: [-90.314% -90.298% -90.265%] (p = 0.00 < 0.05)
                        Performance has improved.
Found 8 outliers among 100 measurements (8.00%)
  5 (5.00%) high mild
  3 (3.00%) high severe

Benchmarking nilike_utf8 scalar contains: Warming up for 3.0000 s
Warning: Unable to complete 100 samples in 5.0s. You may wish to increase target time to 9.1s, enable flat sampling, or reduce sample count to 50.
nilike_utf8 scalar contains
                        time:   [1.7942 ms 1.7962 ms 1.7993 ms]
                        change: [-57.816% -57.765% -57.701%] (p = 0.00 < 0.05)
                        Performance has improved.
Found 6 outliers among 100 measurements (6.00%)
  3 (3.00%) high mild
  3 (3.00%) high severe

nilike_utf8 scalar ends with
                        time:   [243.52 µs 243.58 µs 243.66 µs]
                        change: [-89.761% -89.758% -89.754%] (p = 0.00 < 0.05)
                        Performance has improved.
Found 9 outliers among 100 measurements (9.00%)
  1 (1.00%) low mild
  7 (7.00%) high mild
  1 (1.00%) high severe

nilike_utf8 scalar starts with
                        time:   [253.10 µs 253.18 µs 253.28 µs]
                        change: [-89.367% -89.347% -89.313%] (p = 0.00 < 0.05)
                        Performance has improved.
Found 6 outliers among 100 measurements (6.00%)
  1 (1.00%) high mild
  5 (5.00%) high severe

Benchmarking nilike_utf8 scalar complex: Warming up for 3.0000 s
Warning: Unable to complete 100 samples in 5.0s. You may wish to increase target time to 9.2s, enable flat sampling, or reduce sample count to 50.
nilike_utf8 scalar complex
                        time:   [1.8143 ms 1.8152 ms 1.8162 ms]
                        change: [-5.6406% -5.5148% -5.3303%] (p = 0.00 < 0.05)
                        Performance has improved.
Found 9 outliers among 100 measurements (9.00%)
  4 (4.00%) high mild
  5 (5.00%) high severe

```

# What changes are included in this PR?

# Are there any user-facing changes?

### CODE REVIEW

**Comment by tustvold** [comment]
FYI @askoa

File: arrow-string/src/like.rs [hunk 1]
```diff
@@ -588,50 +588,45 @@ fn ilike_scalar_op<'a, F: Fn(bool) -> bool, L: ArrayAccessor<Item = &'a str>>(
...
-    {
-        // fast path, can use starts_with
-        let start_str = &right[..right.len() - 1].to_uppercase();
-        Ok(BooleanArray::from_unary(left, |item| {
-            op(item.to_uppercase().starts_with(start_str))
-        }))
-    } else if right.starts_with('%') && !right[1..].contains(is_like_pattern) {
-        // fast path, can use ends_with
-        let ends_str = &right[1..].to_uppercase();
+    // If not ASCII faster to use case insensitive regex than allocating using to_uppercase
```
Comment (alamb) [inline arrow-string/src/like.rs:591]:
```suggestion
    // If ASCII faster to use case insensitive regex than allocating using to_uppercase
```

**Review by alamb — COMMENTED** [review]
Do the existing tests cover all the corner cases?

**Comment by tustvold** [comment]
I can add some more, give me a minute

**Comment by tustvold** [comment]
I think this might not work for non-ASCII characters that capitalize to ASCII characters :cry:

File: arrow-string/src/like.rs (hunk 1 above)
Comment (tustvold) [reply arrow-string/src/like.rs:591]: ?

**Comment by alamb** [comment]
Perhaps you can also check that left is entirely ASCII (which is a fairly common case)

**Comment by tustvold** [comment]
Interestingly the regex crate doesn't appear to handle this correctly... :thinking:

**Comment by tustvold** [comment]
Aargh, there are unicode characters that have lowercase ASCII... Guess I will need to check if the string are ASCII

**Comment by tustvold** [comment]
This is currently blocked on https://github.com/apache/arrow-rs/pull/3313, which will allow the kernel to assume a GenericStringArray, which in turn will allow ASCII verification in one pass.

**Comment by tustvold** [comment]
```
ilike_utf8 scalar equals
                        time:   [271.37 µs 271.48 µs 271.62 µs]
                        change: [-87.932% -87.924% -87.914%] (p = 0.00 < 0.05)
                        Performance has improved.
Found 13 outliers among 100 measurements (13.00%)
  2 (2.00%) low mild
  4 (4.00%) high mild
  7 (7.00%) high severe

Benchmarking ilike_utf8 scalar contains: Warming up for 3.0000 s
Warning: Unable to complete 100 samples in 5.0s. You may wish to increase target time to 9.1s, enable flat sampling, or reduce sample count to 50.
ilike_utf8 scalar contains
                        time:   [1.8028 ms 1.8037 ms 1.8048 ms]
                        change: [-56.517% -56.488% -56.455%] (p = 0.00 < 0.05)
                        Performance has improved.
Found 10 outliers among 100 measurements (10.00%)
  5 (5.00%) high mild
  5 (5.00%) high severe

ilike_utf8 scalar ends with
                        time:   [287.38 µs 287.52 µs 287.65 µs]
                        change: [-87.759% -87.745% -87.729%] (p = 0.00 < 0.05)
                        Performance has improved.
Found 3 outliers among 100 measurements (3.00%)
  3 (3.00%) high severe

ilike_utf8 scalar starts with
                        time:   [273.73 µs 273.85 µs 273.98 µs]
                        change: [-88.337% -88.328% -88.318%] (p = 0.00 < 0.05)
                        Performance has improved.
Found 11 outliers among 100 measurements (11.00%)
  1 (1.00%) low mild
  7 (7.00%) high mild
  3 (3.00%) high severe

Benchmarking ilike_utf8 scalar complex: Warming up for 3.0000 s
Warning: Unable to complete 100 samples in 5.0s. You may wish to increase target time to 9.3s, enable flat sampling, or reduce sample count to 50.
ilike_utf8 scalar complex
                        time:   [1.8338 ms 1.8349 ms 1.8361 ms]
                        change: [-4.1219% -3.9358% -3.7411%] (p = 0.00 < 0.05)
                        Performance has improved.
Found 9 outliers among 100 measurements (9.00%)
  4 (4.00%) high mild
  5 (5.00%) high severe

nilike_utf8 scalar equals
                        time:   [257.80 µs 257.85 µs 257.92 µs]
                        change: [-89.265% -89.246% -89.211%] (p = 0.00 < 0.05)
                        Performance has improved.
Found 9 outliers among 100 measurements (9.00%)
  4 (4.00%) high mild
  5 (5.00%) high severe

Benchmarking nilike_utf8 scalar contains: Warming up for 3.0000 s
Warning: Unable to complete 100 samples in 5.0s. You may wish to increase target time to 9.2s, enable flat sampling, or reduce sample count to 50.
nilike_utf8 scalar contains
                        time:   [1.8283 ms 1.8293 ms 1.8303 ms]
                        change: [-57.048% -56.996% -56.926%] (p = 0.00 < 0.05)
                        Performance has improved.
Found 6 outliers among 100 measurements (6.00%)
  2 (2.00%) high mild
  4 (4.00%) high severe

nilike_utf8 scalar ends with
                        time:   [256.38 µs 256.46 µs 256.54 µs]
                        change: [-89.212% -89.192% -89.160%] (p = 0.00 < 0.05)
                        Performance has improved.
Found 8 outliers among 100 measurements (8.00%)
  4 (4.00%) high mild
  4 (4.00%) high severe

nilike_utf8 scalar starts with
                        time:   [262.65 µs 262.70 µs 262.77 µs]
                        change: [-88.974% -88.958% -88.927%] (p = 0.00 < 0.05)
                        Performance has improved.
Found 4 outliers among 100 measurements (4.00%)
  1 (1.00%) high mild
  3 (3.00%) high severe

Benchmarking nilike_utf8 scalar complex: Warming up for 3.0000 s
Warning: Unable to complete 100 samples in 5.0s. You may wish to increase target time to 9.3s, enable flat sampling, or reduce sample count to 50.
nilike_utf8 scalar complex
                        time:   [1.8537 ms 1.8547 ms 1.8559 ms]
                        change: [-3.5514% -3.4042% -3.2212%] (p = 0.00 < 0.05)
                        Performance has improved.
Found 13 outliers among 100 measurements (13.00%)
  2 (2.00%) low mild
  3 (3.00%) high mild
  8 (8.00%) high severe

ilike_utf8_scalar_dyn dictionary[10] string[4])
                        time:   [59.817 µs 59.887 µs 59.968 µs]
                        change: [-97.558% -97.553% -97.546%] (p = 0.00 < 0.05)
                        Performance has improved.
Found 12 outliers among 100 measurements (12.00%)
  5 (5.00%) high mild
  7 (7.00%) high severe
```

So even with the extra check to detect is_ascii, this still is a pretty significant speedup

File: arrow-string/src/like.rs [hunk 2]
```diff
@@ -1272,6 +1270,101 @@ mod tests {
         vec![true, false, false, false]
     );
 
+    // We only implement loose matching
```
Comment (alamb) [inline arrow-string/src/like.rs:1273]: When I ran these tests without the changes in this PR I got a bunch of failures. Is that expected?

```
failures:

---- like::tests::test_utf8_array_ilike_unicode stdout ----
thread 'like::tests::test_utf8_array_ilike_unicode' panicked at 'assertion failed: `(left == right)`
  left: `true`,
 right: `false`: unexpected result when comparing FFkoß at position 0 to FFkoSS ', arrow-string/src/like.rs:1279:5
stack backtrace:
   0: rust_begin_unwind
             at /rustc/897e37553bba8b42751c67658967889d11ecd120/library/std/src/panicking.rs:584:5
   1: core::panicking::panic_fmt
             at /rustc/897e37553bba8b42751c67658967889d11ecd120/library/core/src/panicking.rs:142:14
   2: core::panicking::assert_failed_inner
             at /rustc/897e37553bba8b42751c67658967889d11ecd120/library/core/src/panicking.rs:218:23
   3: core::panicking::assert_failed
             at /rustc/897e37553bba8b42751c67658967889d11ecd120/library/core/src/panicking.rs:181:5
   4: arrow_string::like::tests::test_utf8_array_ilike_unicode
             at ./src/like.rs:1279:5
   5: arrow_string::like::tests::test_utf8_array_ilike_unicode::{{closure}}
             at ./src/like.rs:917:13
   6: core::ops::function::FnOnce::call_once
             at /rustc/897e37553bba8b42751c67658967889d11ecd120/library/core/src/ops/function.rs:248:5
   7: core::ops::function::FnOnce::call_once
             at /rustc/897e37553bba8b42751c67658967889d11ecd120/library/core/src/ops/function.rs:248:5
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.

---- like::tests::test_utf8_array_ilike_unicode_contains stdout ----
thread 'like::tests::test_utf8_array_ilike_unicode_contains' panicked at 'assertion failed: `(left == right)`
  left: `true`,
 right: `false`: unexpected result when comparing sdlkdfFkoßsdfs at position 0 to %FFkoSS% ', arrow-string/src/like.rs:1331:5
stack backtrace:
   0: rust_begin_unwind
             at /rustc/897e37553bba8b42751c67658967889d11ecd120/library/std/src/panicking.rs:584:5
   1: core::panicking::panic_fmt
             at /rustc/897e37553bba8b42751c67658967889d11ecd120/library/core/src/panicking.rs:142:14
   2: core::panicking::assert_failed_inner
             at /rustc/897e37553bba8b42751c67658967889d11ecd120/library/core/src/panicking.rs:218:23
   3: core::panicking::assert_failed
             at /rustc/897e37553bba8b42751c67658967889d11ecd120/library/core/src/panicking.rs:181:5
   4: arrow_string::like::tests::test_utf8_array_ilike_unicode_contains
             at ./src/like.rs:1331:5
   5: arrow_string::like::tests::test_utf8_array_ilike_unicode_contains::{{closure}}
             at ./src/like.rs:917:13
   6: core::ops::function::FnOnce::call_once
             at /rustc/897e37553bba8b42751c67658967889d11ecd120/library/core/src/ops/function.rs:248:5
   7: core::ops::function::FnOnce::call_once
             at /rustc/897e37553bba8b42751c67658967889d11ecd120/library/core/src/ops/function.rs:248:5
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.

---- like::tests::test_utf8_array_ilike_unicode_contains_dyn stdout ----
thread 'like::tests::test_utf8_array_ilike_unicode_contains_dyn' panicked at 'assertion failed: `(left == right)`
  left: `true`,
 right: `false`: unexpected result when comparing sdlkdfFkoßsdfs at position 0 to %FFkoSS% ', arrow-string/src/like.rs:1331:5
stack backtrace:
   0: rust_begin_unwind
             at /rustc/897e37553bba8b42751c67658967889d11ecd120/library/std/src/panicking.rs:584:5
   1: core::panicking::panic_fmt
             at /rustc/897e37553bba8b42751c67658967889d11ecd120/library/core/src/panicking.rs:142:14
   2: core::panicking::assert_failed_inner
             at /rustc/897e37553bba8b42751c67658967889d11ecd120/library/core/src/panicking.rs:218:23
   3: core::panicking::assert_failed
             at /rustc/897e37553bba8b42751c67658967889d11ecd120/library/core/src/panicking.rs:181:5
   4: arrow_string::like::tests::test_utf8_array_ilike_unicode_contains_dyn
             at ./src/like.rs:1331:5
   5: arrow_string::like::tests::test_utf8_array_ilike_unicode_contains_dyn::{{closure}}
             at ./src/like.rs:917:13
   6: core::ops::function::FnOnce::call_once
             at /rustc/897e37553bba8b42751c67658967889d11ecd120/library/core/src/ops/function.rs:248:5
   7: core::ops::function::FnOnce::call_once
             at /rustc/897e37553bba8b42751c67658967889d11ecd120/library/core/src/ops/function.rs:248:5
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.

---- like::tests::test_utf8_array_ilike_unicode_dyn stdout ----
thread 'like::tests::test_utf8_array_ilike_unicode_dyn' panicked at 'assertion failed: `(left == right)`
  left: `true`,
 right: `false`: unexpected result when comparing FFkoß at position 0 to FFkoSS ', arrow-string/src/like.rs:1279:5
stack backtrace:
   0: rust_begin_unwind
             at /rustc/897e37553bba8b42751c67658967889d11ecd120/library/std/src/panicking.rs:584:5
   1: core::panicking::panic_fmt
             at /rustc/897e37553bba8b42751c67658967889d11ecd120/library/core/src/panicking.rs:142:14
   2: core::panicking::assert_failed_inner
             at /rustc/897e37553bba8b42751c67658967889d11ecd120/library/core/src/panicking.rs:218:23
   3: core::panicking::assert_failed
             at /rustc/897e37553bba8b42751c67658967889d11ecd120/library/core/src/panicking.rs:181:5
   4: arrow_string::like::tests::test_utf8_array_ilike_unicode_dyn
             at ./src/like.rs:1279:5
   5: arrow_string::like::tests::test_utf8_array_ilike_unicode_dyn::{{closure}}
             at ./src/like.rs:917:13
   6: core::ops::function::FnOnce::call_once
             at /rustc/897e37553bba8b42751c67658967889d11ecd120/library/core/src/ops/function.rs:248:5
   7: core::ops::function::FnOnce::call_once
             at /rustc/897e37553bba8b42751c67658967889d11ecd120/library/core/src/ops/function.rs:248:5
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.

---- like::tests::test_utf8_array_ilike_unicode_ends stdout ----
thread 'like::tests::test_utf8_array_ilike_unicode_ends' panicked at 'assertion failed: `(left == right)`
  left: `true`,
 right: `false`: unexpected result when comparing sdlkdfFFkoß at position 0 to %FFkoSS ', arrow-string/src/like.rs:1311:5
stack backtrace:
   0: rust_begin_unwind
             at /rustc/897e37553bba8b42751c67658967889d11ecd120/library/std/src/panicking.rs:584:5
   1: core::panicking::panic_fmt
             at /rustc/897e37553bba8b42751c67658967889d11ecd120/library/core/src/panicking.rs:142:14
   2: core::panicking::assert_failed_inner
             at /rustc/897e37553bba8b42751c67658967889d11ecd120/library/core/src/panicking.rs:218:23
   3: core::panicking::assert_failed
             at /rustc/897e37553bba8b42751c67658967889d11ecd120/library/core/src/panicking.rs:181:5
   4: arrow_string::like::tests::test_utf8_array_ilike_unicode_ends
             at ./src/like.rs:924:21
   5: arrow_string::like::tests::test_utf8_array_ilike_unicode_ends::{{closure}}
             at ./src/like.rs:917:13
   6: core::ops::function::FnOnce::call_once
             at /rustc/897e37553bba8b42751c67658967889d11ecd120/library/core/src/ops/function.rs:248:5
   7: core::ops::function::FnOnce::call_once
             at /rustc/897e37553bba8b42751c67658967889d11ecd120/library/core/src/ops/function.rs:248:5
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.

---- like::tests::test_utf8_array_ilike_unicode_ends_dyn stdout ----
thread 'like::tests::test_utf8_array_ilike_unicode_ends_dyn' panicked at 'assertion failed: `(left == right)`
  left: `true`,
 right: `false`: unexpected result when comparing sdlkdfFFkoß at position 0 to %FFkoSS ', arrow-string/src/like.rs:1311:5
stack backtrace:
   0: rust_begin_unwind
             at /rustc/897e37553bba8b42751c67658967889d11ecd120/library/std/src/panicking.rs:584:5
   1: core::panicking::panic_fmt
             at /rustc/897e37553bba8b42751c67658967889d11ecd120/library/core/src/panicking.rs:142:14
   2: core::panicking::assert_failed_inner
             at /rustc/897e37553bba8b42751c67658967889d11ecd120/library/core/src/panicking.rs:218:23
   3: core::panicking::assert_failed
             at /rustc/897e37553bba8b42751c67658967889d11ecd120/library/core/src/panicking.rs:181:5
   4: arrow_string::like::tests::test_utf8_array_ilike_unicode_ends_dyn
             at ./src/like.rs:924:21
   5: arrow_string::like::tests::test_utf8_array_ilike_unicode_ends_dyn::{{closure}}
             at ./src/like.rs:917:13
   6: core::ops::function::FnOnce::call_once
             at /rustc/897e37553bba8b42751c67658967889d11ecd120/library/core/src/ops/function.rs:248:5
   7: core::ops::function::FnOnce::call_once
             at /rustc/897e37553bba8b42751c67658967889d11ecd120/library/core/src/ops/function.rs:248:5
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.

---- like::tests::test_utf8_array_ilike_unicode_starts stdout ----
thread 'like::tests::test_utf8_array_ilike_unicode_starts' panicked at 'assertion failed: `(left == right)`
  left: `true`,
 right: `false`: unexpected result when comparing FFkoßsdlkdf at position 0 to FFkoSS% ', arrow-string/src/like.rs:1291:5
stack backtrace:
   0: rust_begin_unwind
             at /rustc/897e37553bba8b42751c67658967889d11ecd120/library/std/src/panicking.rs:584:5
   1: core::panicking::panic_fmt
             at /rustc/897e37553bba8b42751c67658967889d11ecd120/library/core/src/panicking.rs:142:14
   2: core::panicking::assert_failed_inner
             at /rustc/897e37553bba8b42751c67658967889d11ecd120/library/core/src/panicking.rs:218:23
   3: core::panicking::assert_failed
             at /rustc/897e37553bba8b42751c67658967889d11ecd120/library/core/src/panicking.rs:181:5
   4: arrow_string::like::tests::test_utf8_array_ilike_unicode_starts
             at ./src/like.rs:1291:5
   5: arrow_string::like::tests::test_utf8_array_ilike_unicode_starts::{{closure}}
             at ./src/like.rs:917:13
   6: core::ops::function::FnOnce::call_once
             at /rustc/897e37553bba8b42751c67658967889d11ecd120/library/core/src/ops/function.rs:248:5
   7: core::ops::function::FnOnce::call_once
             at /rustc/897e37553bba8b42751c67658967889d11ecd120/library/core/src/ops/function.rs:248:5
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.

---- like::tests::test_utf8_array_ilike_unicode_start_dyn stdout ----
thread 'like::tests::test_utf8_array_ilike_unicode_start_dyn' panicked at 'assertion failed: `(left == right)`
  left: `true`,
 right: `false`: unexpected result when comparing FFkoßsdlkdf at position 0 to FFkoSS% ', arrow-string/src/like.rs:1291:5
stack backtrace:
   0: rust_begin_unwind
             at /rustc/897e37553bba8b42751c67658967889d11ecd120/library/std/src/panicking.rs:584:5
   1: core::panicking::panic_fmt
             at /rustc/897e37553bba8b42751c67658967889d11ecd120/library/core/src/panicking.rs:142:14
   2: core::panicking::assert_failed_inner
             at /rustc/897e37553bba8b42751c67658967889d11ecd120/library/core/src/panicking.rs:218:23
   3: core::panicking::assert_failed
             at /rustc/897e37553bba8b42751c67658967889d11ecd120/library/core/src/panicking.rs:181:5
   4: arrow_string::like::tests::test_utf8_array_ilike_unicode_start_dyn
             at ./src/like.rs:1291:5
   5: arrow_string::like::tests::test_utf8_array_ilike_unicode_start_dyn::{{closure}}
             at ./src/like.rs:917:13
   6: core::ops::function::FnOnce::call_once
             at /rustc/897e37553bba8b42751c67658967889d11ecd120/library/core/src/ops/function.rs:248:5
   7: core::ops::function::FnOnce::call_once
             at /rustc/897e37553bba8b42751c67658967889d11ecd120/library/core/src/ops/function.rs:248:5
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.


failures:
    like::tests::test_utf8_array_ilike_unicode
    like::tests::test_utf8_array_ilike_unicode_contains
    like::tests::test_utf8_array_ilike_unicode_contains_dyn
    like::tests::test_utf8_array_ilike_unicode_dyn
    like::tests::test_utf8_array_ilike_unicode_ends
    like::tests::test_utf8_array_ilike_unicode_ends_dyn
    like::tests::test_utf8_array_ilike_unicode_start_dyn
    like::tests::test_utf8_array_ilike_unicode_starts
```

Comment (tustvold) [reply arrow-string/src/like.rs:1273]: Yes see #3311

**Review by alamb — APPROVED** [review]
LGTM -- cc @Dandandan

**Pull request merged**